pkg/logger/dapr_logger.go

```diff
@@ -0,0 +1,195 @@
+package logger
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/dapr/dapr/pkg/version"
+	"github.com/dapr/dapr/utils"
+	"github.com/sirupsen/logrus"
+)
+
+var daprLogLevelToLogrusLevel = map[LogLevel]logrus.Level{
```

Comment:
Can we remove this map? this file (`dapr_logger.go`) has references to `logrus` and uses `logrus` methods, so you can safely use `logrus.ParseLevel` here in `SetOutputLevel()`.

Reply:
I understand what you want. It requires type casting from LogLevel to string again to use logrus.ParseLevel(). As mentioned, I am not a fan of casting. I prefer map or mapping function to convert from TypeA to TypeB.

It is too late. let's talk on monday in person.

Reply:
Actually, there's no casting required. Because this is a string type, you can simply format it:

```go
info := InfoLevel
l, err := logrus.ParseLevel(fmt.Sprint(info))
```

Reply:
If I need to use fmt.Sprint, then I would use casting. 

[Logrus ParseLevel implementation](https://github.com/sirupsen/logrus/blob/947831125f318c2fb34bfc6205ee5d74994deb88/logrus.go#L25-L45) is just mapping function. 

I still think using our own mapping is clear and straight-forward rather than casting and converting to string via fmt.sprint.

Reply:
I used the casting here. please review again.

Reply:
I had a bad experience when I was developing Application Insight Pipeline. After that, I prefer the explicit mapping if the code is OSS. I created toLogrusLevel using logrus.ParseLevel and added unit-tests to ensure the expected conversion.

Reply:
> I used the casting here. please review again.

Just fyi, there's no casting in Go. Using `string(X)` is called explicit conversion, and is much safer to use. Casting as done in C# for example is not possible in Go due to its type system. The closest you can get is with type assertions (`x.(y)`), but that's for very constrained use cases.

I myself have bad experiences with mappings, for reasons including ambiguity of intent and maintainability.